### PR TITLE
refactor(lsp): add deprecate in  _resolve_capabilities_compat

### PR DESCRIPTION
**Pull request by glepnir**
this function is never used now. mark as deprecate in 0.10 .

### CODE REVIEW

File: runtime/lua/vim/lsp/protocol.lua [hunk 1]
```diff
@@ -898,6 +898,7 @@ end
 ---@param server_capabilities table Table of capabilities supported by the server
 ---@return table Normalized table of capabilities
 function protocol._resolve_capabilities_compat(server_capabilities)
+  vim.deprecate('protocal_resolve_capabilities', nil, '0.10')
```
Comment (nullchilly) [inline runtime/lua/vim/lsp/protocol.lua:901]: Shouldn't the version be `'1.0'`?

Comment (clason) [reply runtime/lua/vim/lsp/protocol.lua:901]: no!

Comment (clason) [reply runtime/lua/vim/lsp/protocol.lua:901]: (after `0.9.x` comes `0.10.0`, and then on until `0.100.0`; `1.0.0` has a very specific meaning (stability) which we are still some ways away from.)

**Review by mfussenegger — COMMENTED** [review]
Given that this is both marked as `@private` and prefixed with `_` to indicate it's private I think we can just remove this?

**Comment by glepnir** [comment]
just search on github also no this method used. so remove is safe maybe .

**Comment by clason** [comment]
> just search on github also no this method used. so remove is safe maybe .

If someone on Github used a function clearly marked "private", they'll get what's coming to them (and are hopefully prepared for it). There's no reason to take this into consideration.

**Comment by clason** [comment]
Feels much better with these diffstats, doesn't it ;)

**Pull request merged**